File: app/src/main/java/com/jiujiu/mvvmTemplate/ui/orders/OrdersFragment.java
package com.jiujiu.mvvmTemplate.ui.orders;


import android.os.Bundle;
import android.util.Log;
import android.view.View;

import com.jiujiu.mvvmTemplate.BR;
import com.jiujiu.mvvmTemplate.R;
import com.jiujiu.mvvmTemplate.databinding.FragmentOrdersBinding;
import com.jiujiu.mvvmTemplate.ui.base.BaseFragment;


public class OrdersFragment extends BaseFragment<FragmentOrdersBinding, OrdersFragViewModel> {
    private static final String TAG = "OrdersFragment";
    private static OrdersFragment mFragment;

    public OrdersFragment() {
        Log.d(TAG, "OrdersFragment: constructors");
    }

    public static OrdersFragment getInstance() {
        if (mFragment == null) {
            mFragment = new OrdersFragment();
        }
        return mFragment;
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        setupViewModel();
        Log.d(TAG, "onViewCreated: ");
    }

    private void setupViewModel() {
    }

    @Override
    protected int getBindingVariableId() {
        return BR.viewModel;
    }

    @Override
    protected int getLayoutId() {
        return R.layout.fragment_orders;
    }

    @Override
    protected Class<OrdersFragViewModel> getViewModelType() {
        return OrdersFragViewModel.class;
    }

}

File: app/src/main/java/com/jiujiu/mvvmTemplate/ui/orders/OrdersFragment.kt
package com.jiujiu.mvvmTemplate.ui.orders


import android.os.Bundle
import android.view.View
import com.jiujiu.mvvmTemplate.BR
import com.jiujiu.mvvmTemplate.R
import com.jiujiu.mvvmTemplate.databinding.FragmentOrdersBinding
import com.jiujiu.mvvmTemplate.ui.base.BaseFragment
import org.jetbrains.anko.info


class OrdersFragment : BaseFragment<FragmentOrdersBinding, OrdersFragViewModel>() {

    override val bindingVariableId: Int
        get() = BR.viewModel

    override val layoutId: Int
        get() = R.layout.fragment_orders

    override val viewModelType: Class<OrdersFragViewModel>
        get() = OrdersFragViewModel::class.java


    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        setupViewModel()
        info("onViewCreated: ")
    }

    private fun setupViewModel() {}

    companion object {
        fun getInstance(): OrdersFragment = OrdersFragment()
    }

}
